diff --git a/dist/vanilla-extract-integration.cjs.dev.js b/dist/vanilla-extract-integration.cjs.dev.js
index 6048f011f091fe2a9f88886e8a1a839c1fcad39f..3e69d44e83ab53602987f591d00ac3f40ec7877a 100644
--- a/dist/vanilla-extract-integration.cjs.dev.js
+++ b/dist/vanilla-extract-integration.cjs.dev.js
@@ -400,6 +400,18 @@ const vanillaExtractTransformPlugin = ({
 
   setup(build) {
     const packageInfo = getPackageInfo(build.initialOptions.absWorkingDir);
+
+    build.onLoad(
+      {filter: /.vanilla.css.ts$/}, 
+      async ({path: path$1}) => {
+      const css = await fs.promises.readFile(path$1, 'utf-8');
+      return {
+        contents: css,
+        loader: 'text',
+        resolveDir: path.dirname(path$1)
+      }
+    });
+
     build.onLoad({
       filter: cssFileFilter
     }, async ({
diff --git a/dist/vanilla-extract-integration.cjs.prod.js b/dist/vanilla-extract-integration.cjs.prod.js
index 082e846c8a9f90adfa0a03b816867b9e8d2b59ed..33354cc332873ee225d29caaf3ef16598a1537ef 100644
--- a/dist/vanilla-extract-integration.cjs.prod.js
+++ b/dist/vanilla-extract-integration.cjs.prod.js
@@ -400,6 +400,18 @@ const vanillaExtractTransformPlugin = ({
 
   setup(build) {
     const packageInfo = getPackageInfo(build.initialOptions.absWorkingDir);
+
+    build.onLoad(
+      {filter: /.vanilla.css.ts$/}, 
+      async ({path: path$1}) => {
+      const css = await fs.promises.readFile(path$1, 'utf-8');
+      return {
+        contents: css,
+        loader: 'text',
+        resolveDir: path.dirname(path$1)
+      }
+    });
+
     build.onLoad({
       filter: cssFileFilter
     }, async ({